### PR TITLE
ENH: Remove unnecessary restriction in noncen-f

### LINKED ISSUE #6638: Improve random variate generation of noncentral F-distribution
The noncentral F-distribution has three parameters: 
- `dfnum`, degrees of freedom in numerator
- `dfden`, degrees of freedom in denominator 
- `nonc`, noncentrality 

According to [http://docs.scipy.org/doc/numpy/reference/generated/numpy.random.noncentral_f.html](http://docs.scipy.org/doc/numpy/reference/generated/numpy.random.noncentral_f.html), 
the current implementation unnecessarily requires both `dfnum` and `dfden` be greater than 1, while they are allowed to be any positive value in theory. 

This restriction stems from the fact that the implementation calls `random.noncentral_chisquare`, which unnecessarily required that its `dof` argument must be greater than 1 and has been fixed in `numpy 1.10`; see #5766. 

Since the implementation of `random.noncentral_f` is straightforward, something like 

``` python
t = np.random.noncentral_chisquare(dfnum, nonc) * dfden 
t / (np.random.chisquare(dfden) * dfnum)
```

according to [https://github.com/numpy/numpy/blob/master/numpy/random/mtrand/distributions.c#L256-L260 ](https://github.com/numpy/numpy/blob/master/numpy/random/mtrand/distributions.c#L256-L260), this should be an easy fix.

### PR DESCRIPTION
**Pull request by bashtage**
Set dfnum restriction to be > 0 as required by noncentral chisquare

closes #6638

### CODE REVIEW

File: numpy/random/mtrand/mtrand.pyx [hunk 1]
```diff
@@ -2175,16 +2175,16 @@ cdef class RandomState:
...
             if fdfden <= 0:
                 raise ValueError("dfden <= 0")
             if fnonc < 0:
                 raise ValueError("nonc < 0")
             return cont3_array_sc(self.internal_state, rk_noncentral_f, size,
                                   fdfnum, fdfden, fnonc, self.lock)
 
-        if np.any(np.less_equal(odfnum, 1.0)):
+        if np.any(np.less_equal(odfnum, 0.0)):
             raise ValueError("dfnum <= 1")
```
Comment (eric-wieser) [inline numpy/random/mtrand/mtrand.pyx]: Message needs changing too

**Comment by charris** [comment]
Thanks @bashtage . I note that both chisquare and noncentral_chisquare document degrees of freedom to be ints, which is not necessarily the case. Might be worth fixing that and checking that other chisquare derived distributions are correct.

**Comment by xiaoweiz** [comment]
Hi, @bashtage 
Thanks for taking time fixing the issue. However, as of version 1.13.3, the bug is still there. (I've tried the installation through both Conda and pip.)
image

**Comment by bashtage** [comment]
@xiaoweiz It won't be out until 1.14 since it is an enhancement, not a bug fix.

**Comment by xiaoweiz** [comment]
Thanks!

**Pull request merged**